### ISSUE TITLE
ValueProxySource Crashes Resonite Completely

### ISSUE THREAD

**Issue by AmasterAmaster** [Yellow-Dog-Man/Resonite-Issues#2694]
### Describe the bug?

This one was found by accident when figuring out on how to use AvatarEquipBlock in different ways to access it's fields. When my friend decided to use the ValueProxySource as a way to transfer over a component... then boom! Not only did the world crash, the entire Resonite application did too.

I narrowed it down to ValueProxySource, as it does not like any other type than values types. Putting a reference type in there and attaching it to any slot will make it crash Resonite.

### To Reproduce

Here is how you do it:
* Spawn an Empty Object
* Use the component browser and search for ValueProxySource (Located in UIX -> Interaction -> ValueProxySource)
* Select any type that is **NOT** a value type (I used `AvatarEquipBlock` type, and for fun I even used the `If` type)
* Double click it to attach it and watch as Resonite freezes. With the unfortunate effect that log files will be cut off, preventing one from getting the error... unless a friend did this for you (See "Additional Context" for more info)

### Expected behavior

The ValueProxySource should not give you the option to select and attach reference types for this component.

### Screenshots

This (e) is a deadly weapon:

![This is a deadly weapon 01](https://github.com/user-attachments/assets/97962883-f74e-46cf-87ad-144bef06be00)

And yes, I can confirm that this crashes too.

### Resonite Version Number

Beta 2024.7.25.1284

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Index & Desktop

### Log Files

[ANDREW-SUPERPC - 2024.7.25.1284 - 2024-07-31 02_29_50.log](https://github.com/user-attachments/files/16438073/ANDREW-SUPERPC.-.2024.7.25.1284.-.2024-07-31.02_29_50.log)

### Additional Context

I suspect that it has to do with this error (Taken from another log file when a friend did this first):
`System.TypeLoadException: Could not resolve type with token 01000022 (from typeref, class/assembly SkyFrost.Base.Contact, SkyFrost.Base, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null)`

I am unsure if this will help, as I can't confirm this, but it is the only lead I got currently.

### Reporters

AmasterAmaster
Decoy

**Comment by AmasterAmaster** [comment]
@charlie-sans That might be a more accurate lead, yes

**Comment by AmasterAmaster** [comment]
It also seems that other types are fine (stating from the most sane until I hit the crash again), such as : `float`, `decimal`, then to `IField` (Surprised me that it did not crash from that), `IWorldElement`, `RefID`. These seem fine.

But the moment I put in `Slot`, it hates me. XD

(My version of sanity checking things.)

**Comment by charlie-sans** [comment]
makes sense, most of those shouldn't crash you but seeing that even before what seems to be the crash, that they are logging even your radial menu plus some slots after that makes me wonder whats going on. did the program hit a stutter then print out a stack trace of the scene or is this just some long forgot logging code lol

**Comment by Nytra** [comment]
The radial menu logging is normal it happens all the time

**Comment by Nytra** [comment]
This is most likely related or possibly a duplicate of this issue: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1646

**Comment by AmasterAmaster** [comment]
Did not know it was mentioned in the other issue, damn. But is the method of doing it the same? As it is as easy as just attaching a component.

**Comment by shiftyscales** [comment]
> But is the method of doing it the same?

Yes. As I mentioned in this comment https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1646#issuecomment-2043657755 I replicated it with the Slot reference type previously.
